client/boot/index.js

```diff
@@ -217,6 +218,18 @@ function reduxStoreReady( reduxStore ) {
 		renderLayout( reduxStore );
 	}
 
+	const isLegacyRoute = ( path ) => {
+		return ( /.php$/.test( path ) ||
+			/^\/?$/.test( path ) && ! config.isEnabled( 'reader' ) ||
+			/^\/my-stats/.test( path ) ||
+			/^\/notifications/.test( path ) ||
+			/^\/themes/.test( path ) ||
+			/^\/manage/.test( path ) ||
+			/^\/plans/.test( path ) && ! config.isEnabled( 'manage/plans' ) ||
+			/^\/me/.test( path ) && ! /^\/me\/billing/.test( path ) &&
+			! /^\/me\/next/.test( path ) && ! config.isEnabled( 'me/my-profile' ) );
+	};
```

Comment:
Of course, with all the repetition here and mixed data and logic, it seems ripe for some helper function…

```js
const notEnabled = feature => () => ! config.isEnabled( feature );

const matchingMePaths = path =>
	! /^\/me\/billing/.test( path ) &&
	! /^\/me\/next/.test( path ) &&
	notEnabled( 'me/my-profile' );

// [ [ path matching Regex, optional predicate condition ] ]
const legacyRoutes = [
	[ /.php&/ ],
	[ /^\/?$/, notEnabled( 'reader' ],
	[ /^\/my-stats/ ],
	[ /^\/notifications/ ],
	[ /^\/themes/ ],
	…
	[ /^\/me/, matchingMePaths ]
];

const isLegacyRoute = path => 
	legacyRoutes.some( ( [ pattern, predicate = constant( true ) ] => pattern.test( path ) && predicate( path ) );
```